app/src/main/java/org/simple/clinic/search/results/PatientSearchResultsController.kt

```diff
@@ -74,14 +74,8 @@ class PatientSearchResultsController @Inject constructor(
   }
 
   private fun saveEntryAndGoToRegisterPatientScreen(ongoingNewPatientEntry: OngoingNewPatientEntry, currentFacility: Facility): Observable<UiChange> {
-    val nextScreenChange = if (isFacilitySwitchedPreference.get()) {
-      { ui: Ui -> ui.openAlertFacilityChangeSheet(currentFacility) }
-    } else {
-      { ui: Ui -> ui.openPatientEntryScreen() }
-    }
-
     return patientRepository
         .saveOngoingEntry(ongoingNewPatientEntry)
-        .andThen(Observable.just(nextScreenChange))
+        .andThen(Observable.just { ui: Ui -> ui.openAlertFacilityChangeSheet(currentFacility) })
```

Comment:
I think we can continue calling `openPatientEntryScreen()` here and leave it up to a UI implementation to decide whether to invoke the facility change sheet or not. What do you think?

Reply:
Shouldn't this be part of the business logic and not the UI?
Also, it can be misleading.

Reply:
The way I am seeing it is that the business logic is to open the patient entry screen and that confirming the facility switch is something that is left up the UI to decide as an interception.

But if you would like to leave it like this, I am fine with this too.

Reply:
Yes. I understand what you mean. I will make this change.